nifi-nar-bundles/nifi-standard-services/nifi-hbase-client-service-api/src/main/java/org/apache/nifi/hbase/HBaseClientService.java

```diff
@@ -138,6 +169,19 @@
      */
     void scan(String tableName, Collection<Column> columns, String filterExpression, long minTime, ResultHandler handler) throws IOException;
 
+    /**
+     * Scans the given table using the optional filter criteria and passing each result to the provided handler.
+     *
+     * @param tableName the name of an HBase table to scan
+     * @param columns optional columns to return, if not specified all columns are returned
+     * @param filterExpression optional filter expression, if not specified no filtering is performed
+     * @param minTime the minimum timestamp of cells to return, passed to the HBase scanner timeRange
+     * @param authorizations the visibility labels to apply to the scanner.
+     * @param handler a handler to process rows of the result set
+     * @throws IOException thrown when there are communication errors with HBase
+     */
+    void scan(String tableName, Collection<Column> columns, String filterExpression, long minTime, List<String> authorizations, ResultHandler handler) throws IOException;
```

Comment:
I believe the argument name for the `List<String>` should be 'visibilityLabels' at this point. `org.apache.hadoop.hbase.security.visibility.Authorizations` is created later by using these labels. The same comment goes for other overloaded scan methods.

Reply:
I originally screwed up the naming here, and adopted a more Accumulo-like convention to make it easier. Authorizations are a flat list of tokens, labels are the complex boolean string that is evaluated with those authorizations during the scan.